Use gds-api-adapters for publishing-api requests

tijmenb opened the pull request:
This allows us to use the bearer token, which is needed in integration.

benhyland commented:
If we're introducing the Services module pattern, can we use it for Rummager as well?

tijmenb commented:
Yes, Ben. You may do that!

benhyland commented:
In this PR? :P

tijmenb committed:
Use gds-api-adapters for publishing-api requests

This allows us to use the bearer token, which is needed in integration.

Also uses the `Services` module for all calls to external APIs, which
is the pattern for most apps on GOV.UK.

tijmenb commented:
Why not. Updated & rebased.
